tools/clang/test/CodeGenSPIRV/cast.struct-to-int.hlsl

```diff
@@ -83,22 +83,21 @@ void main()
 // CHECK: [[COLORS:%[^ ]*]] = OpLoad [[TWOCOLORS]]
 // CHECK: [[COLORS0:%[^ ]*]] = OpCompositeExtract [[COLORRGBA]] [[COLORS]] 0
 // CHECK: [[COLORS00:%[^ ]*]] = OpCompositeExtract [[UINT]] [[COLORS0]] 0
-// CHECK: [[COLORS000:%[^ ]*]] = OpBitFieldUExtract [[UINT]] [[COLORS00]] [[U0]] [[U8]]
 // CHECK: [[BUF00:%[^ ]*]] = OpAccessChain %{{[^ ]*}} [[BUF]] [[I0]] [[U0]]
-// CHECK: OpStore [[BUF00]] [[COLORS000]]
+// CHECK: OpStore [[BUF00]] [[COLORS00]]
```

Comment:
I looked at the DXIL code, and this seems to be the correct behaviour. Could someone else check? We assign the scalar to the bit container instead of to the individual bit fields.

Reply:
Seems to me that this is correct.
The DXIL code is a bit weird, they load 3 times the single uint, maybe because a code part sees 3 fields, but overall, the generated SPIR-V code seems to match: the whole uint is stored in the array.

Reply:
I think my implementation is correct for this case. I tried another test (https://godbolt.org/z/PPjGWfcKG), and I think I have that one "wrong". I personally think the DXIL behaviour is incomprensible.

It seems like each bit field will count for a different scalar  when flattening, but the value of the scalar is not the value of the bitfield. It is the value of the contain holding the bitfiled. So in the new test case, `buf[0]` gets every element set to the same value `524255231` not `{{255, 127}, {63, 31}}` as you might expect.

Reply:
I added that example as a test, and made it work.